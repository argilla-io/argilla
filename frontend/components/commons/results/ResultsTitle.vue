<template>
  <p class="records-title" v-if="dataset.results.total > 0">
    Records
    <span v-if="showWhenFiltered && areFiltersApplied.length"
      >with filters applied</span
    >
    ({{ dataset.results.total | formatNumber }})
  </p>
</template>

<script>
export default {
  props: {
    dataset: {
      type: Object,
      required: true,
    },
    showWhenFiltered: {
      type: Boolean,
      required: false,
    },
    excludedFilter: {
      type: Array,
      default: () => {
        return ["text"];
      },
    },
  },
  computed: {
    areFiltersApplied() {
      const appliedFilters = Object.keys(this.dataset.query)
        .filter((f) => !this.excludedFilter.includes(f))
        .map((key) => this.dataset.query[key]);
      return appliedFilters.filter((v) => v && Object.values(v).length);
    },
  },
};
</script>

<style lang="scss" scoped>

.records-title {
  color: $font-secondary;
}
</style>
