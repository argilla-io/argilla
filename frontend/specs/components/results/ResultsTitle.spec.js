import { mount } from "@vue/test-utils";
import ResultsTitle from "@/components/commons/results/ResultsTitle";

function mountResultsTitle() {
  return mount(ResultsTitle, {
    propsData: {
      dataset: {
        task: "TextClassification",
        results: {
          total: 400,
        },
        query: {
          annotated_as: undefined,
          annotated_by: undefined,
          from: undefined,
          limit: undefined,
          metadata: {},
          predicted: undefined,
          predicted_as: ["Alta"],
          predicted_by: undefined,
          score: undefined,
          status: undefined,
          text: "de",
        },
      },
      showWhenFiltered: true,
      excludedFilter: ["text"],
    },
  });
}

describe("ResultsTitle", () => {
  let spy = jest.spyOn(console, "error");
  afterEach(() => spy.mockReset());

  test("renders properly", () => {
    const wrapper = mountResultsTitle();
    expect(wrapper).toMatchSnapshot();
  });
});
