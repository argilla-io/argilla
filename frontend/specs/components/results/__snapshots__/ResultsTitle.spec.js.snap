// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsTitle renders properly 1`] = `
<p class="records-title">
  Records
  <span>with filters applied</span>
  (400)
</p>
`;
